### PR TITLE
For max_pool_with_argmax op, GPU kernel keeps consistent with the behavior of CPU

### LINKED ISSUE #22025: [Bug] max_pool_with_argmax has different behaviour on CPU and GPU
The function ``max_pool_with_argmax`` has different behaviour concerning the returned indices depending on the GPU or CPU backend.

- On GPU, returned indices do not take into account batch dimension. 
It returns indices such that `[b, y, x, c] -> (y * width + x) * channels + c `
- On CPU, returned indices take into account batch dimension.
It returns indices such that `[b, y, x, c] -> ((b * height + y) * width + x) * channels + c `

Isn't it a problem if one wants to run models with the same graph including this op both on CPU and GPU ?

Here is an example to show the problem:
``` python
import numpy as np
import tensorflow as tf 

i = tf.placeholder(tf.float32, [None, 4, 4, 1])
p, ind = tf.nn.max_pool_with_argmax(i, ksize=[1, 2, 2, 1], strides=[1, 2, 2, 1], padding='SAME')

with tf.Session() as sess:
    a = np.random.randint(10, size=(2, 4, 4, 1))
    print a[:,:,:,0]
    print sess.run(p, {i: a})[:,:,:,0]
    print sess.run(ind, {i: a})[:,:,:,0]
```
Which results in something like this on CPU:
```
[[[0 3 0 0]
  [4 0 4 8]
  [7 8 4 5]
  [2 3 4 5]]

 [[2 5 2 7]
  [4 3 4 6]
  [8 3 2 0]
  [7 5 9 3]]]
[[[4. 8.]
  [8. 5.]]

 [[5. 7.]
  [8. 9.]]]
[[[ 4  7]
  [ 9 11]]

 [[17 19]
  [24 30]]]
```
And something like this on GPU:
```
[[[8 3 5 5]
  [0 0 0 3]
  [6 9 4 1]
  [2 0 8 8]]

 [[4 6 6 5]
  [5 7 9 3]
  [7 6 1 3]
  [1 6 1 0]]]
[[[8. 5.]
  [9. 8.]]

 [[7. 9.]
  [7. 3.]]]
[[[ 0  2]
  [ 9 14]]
 [[ 5  6]
  [ 8 11]]]
```
This was obtained with 1.9 CPU/GPU tensorflow versions.
I think it would be more convenient to have the same behaviour on CPU and GPU, isn't it ?

### PR DESCRIPTION
**Pull request by facaiy**
Fix #22025

Respect with [ tf.nn.max_pool_with_argmax: API](https://www.tensorflow.org/api_docs/python/tf/nn/max_pool_with_argmax) , now GPU implementation returns flattened index `((b * height + y) * width + x) * channels + c`.

### CODE REVIEW

**Comment by facaiy** [comment]
@poxvoculi  @rmlarsen Hi, could you take a look or reassign? Thanks.

**Comment by facaiy** [comment]
I checked those failed checks. Seems unrelated?

**Comment by facaiy** [comment]
@yifeif Hi, yifei. Could you take a look? Thank you :-)

**Comment by yifeif** [comment]
Thanks @facaiy!  @rmlarsen would you be the right person to review changes to maxpooling_op?

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rmlarsen: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rmlarsen: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rmlarsen: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 44 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @rmlarsen: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 59 days with no activity and the `awaiting review` label has been applied.

**Comment by facaiy** [comment]
Close it since #26562 is created to fix the issue.
